### ISSUE TITLE
segfault with left join and inclusive ranges against tables w/ timestamps in 0.3.3

### ISSUE THREAD

**Issue by hamilton** [duckdb/duckdb#3486]
#### What happens?

I think this is a new bug, since we have in our application a nearly identical query that is working in 0.3.2.

- Essentially, for some reason the `LEFT JOIN` with an inclusive range in the reproducing steps below is causing a `SEGFAULT` in both node.js and the CLI. 
- One can construct a similar query with a `DOUBLE` (in other words, no timestamp in a source table) and it works as-expected – below, I'm using `epoch` to get the timestamp into posix time so I can bin as if it were numeric.
- One can additionally construct a similar set of queries with a **view** w/ timestamps, and it appears to work.

So there is something about a table with a timestamp column causing a segfault issue with this particular join that is new to 0.3.3.

I can't try this on bleeding edge CLI since I'm getting a 404 when I try to download it. I can reproduce it with the node.js lib on `@next` however.

This issue is blocking us from migrating to 0.3.3.

#### To Reproduce

Run this script in the CLI:

```sql
CREATE TABLE tbl AS 
SELECT generate_series as ts from generate_series(timestamp '2020-01-01', timestamp '2021-01-01', interval 1 day);
WITH data_table AS (
    SELECT epoch(ts) as ts 
    FROM tbl
    WHERE ts IS NOT NULL
  ), 
  S AS (
    SELECT 
      min(ts) as minVal,
      max(ts) as maxVal,
      (max(ts) - min(ts)) as range
      FROM data_table
  ), 
  buckets AS (
    SELECT
      range as bucket,
      (range) * (select range FROM S) / 40 + (select minVal from S) as low,
      (range + 1) * (select range FROM S) / 40 + (select minVal from S) as high
    FROM range(0, 40, 1)
  )
  SELECT
      bucket,
      low,
      high,
      count(data_table.ts) as count
    FROM buckets
    LEFT JOIN data_table ON (data_table.ts >= low AND data_table.ts < high)
    GROUP BY bucket, low, high
    ORDER BY bucket;
```

If you replace the `CREATE TABLE` with `CREATE VIEW`, the error disappears and a resultset is generated.

Sorry I can't be more helpful here. I'm a bit mystified by this but attempted to create a reproducible example based on my understanding of the problem.

#### Environment (please complete the following information):
 - OS: Mac 12.1 Monterey (M1)
 - DuckDB Version: 0.3.3
 - DuckDB Client: CLI, Node.js

#### Before Submitting

- [x] **Have you tried this on the latest `master` branch?** – yes, but only for Node.js, which reproduces this issue.
* **Python**: `pip install duckdb --upgrade --pre`
* **R**: `install.packages("https://github.com/duckdb/duckdb/releases/download/master-builds/duckdb_r_src.tar.gz", repos = NULL)`
* **Other Platforms**: You can find binaries [here](https://github.com/duckdb/duckdb/releases/tag/master-builds) or compile from source.

- [x] **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?**

**Comment by hawkfish** [comment]
Sigh, this is a dumb mistake in the outer IEJoin code. It assumes that all outer blocks have missing matches. Easy to fix.

**Comment by Mytherin** [comment]
Fixed by #3491
